blocks/library/button/index.js

```diff
@@ -101,24 +102,34 @@ class ButtonBlock extends Component {
 							checked={ !! clear }
 							onChange={ this.toggleClear }
 						/>
-						<PanelColor title={ __( 'Background Color' ) } colorValue={ color } >
-							<ColorPalette
-								value={ color }
-								onChange={ ( colorValue ) => setAttributes( { color: colorValue } ) }
-							/>
-						</PanelColor>
-						<PanelColor title={ __( 'Text Color' ) } colorValue={ textColor } >
-							<ColorPalette
-								value={ textColor }
-								onChange={ ( colorValue ) => setAttributes( { textColor: colorValue } ) }
-							/>
-						</PanelColor>
-						{ this.nodeRef && <ContrastCheckerWithFallbackStyles
-							node={ this.nodeRef }
-							textColor={ textColor }
-							backgroundColor={ color }
-							isLargeText={ true }
-						/> }
+						<ColorContext>
+							{ ( { colors, disableCustomColors, hasColorsToChoose } ) => ( hasColorsToChoose ? [
```

Comment:
It might make sense to refactor this code to work with `Fragment` component instead of an array:
```js
{ ( { colors, disableCustomColors, hasColorsToChoose } ) => ( hasColorsToChoose && (
    <Fragment>
        <PanelColor title={ __( 'Background Color' ) } colorValue={ color }>
        ....
    </Fragment>
) }
```

Reply:
I always question my self about what to use arrays or fragment component. For this cases, I will try to default to Fragment from now on. The code was updated.